apps/src/music/player/ToneJSPlayer.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * TODO:
+ * - sequence playback
+ * - Effects
+ */
+
+import {SoundLoadCallbacks} from '../types';
+import SoundCache from './SoundCache';
+
+const DEFAULT_BPM = 120;
+
+export interface SampleEvent {
+  id: string;
+  transportTime: string;
+  originalBpm: number;
+  pitchShift: number;
+}
+
+export interface SamplerSequence {
+  instrument: string;
+  events: {notes: string[]; transportTime: string}[];
+}
+
+class ToneJSPlayer {
+  private samplers: {[instrument: string]: typeof Tone.Sampler};
+  private activePlayers: (typeof Tone.GrainPlayer)[];
+  private currentPreview: {id: string; player: typeof Tone.GrainPlayer} | null;
+
+  constructor(
+    bpm = DEFAULT_BPM,
+    private readonly soundCache: SoundCache = new SoundCache()
+  ) {
+    Tone.Transport.bpm.value = bpm;
+    this.activePlayers = [];
+    this.samplers = [];
+    this.currentPreview = null;
+  }
+
+  getCurrentPosition(): string {
+    return Tone.Transport.position;
+  }
+
+  goToPosition(position: string) {
+    Tone.Transport.position = position;
+  }
+
+  setLoopEnabled(enabled: boolean) {
+    Tone.Transport.loop = enabled;
+  }
+
+  setLoopStart(startPosition: string) {
+    Tone.Transport.loopStart = startPosition;
+  }
+
+  setLoopEnd(endPosition: string) {
+    Tone.Transport.loopEnd = endPosition;
+  }
+
+  async loadSounds(sampleIds: string[], callbacks?: SoundLoadCallbacks) {
+    return this.soundCache.loadSounds(sampleIds, callbacks);
+  }
+
+  async loadInstrument(
+    instrumentName: string,
+    sampleMap: {[note: number]: string},
+    callbacks?: SoundLoadCallbacks
+  ) {
+    const urls: {[note: number]: AudioBuffer} = {};
```

Comment:
question.. do you need to load the instrument if previously loaded?

Also, is this the only reference to the instrument object? I was thinking this could be a good idea
```
    if (this.samplers[instrumentName]) {
      this.samplers[instrumentName].dispose();
    }
```